src/workerd/api/sql.c++

```diff
@@ -286,4 +502,44 @@ void SqlStorage::visitForMemoryInfo(jsg::MemoryTracker& tracker) const {
   }
 }
 
+jsg::Ref<SqlStorage::Cursor> SqlStorage::BoundStatement::run() {
+  return statement->run(cloneBindings());
+}
+
+jsg::JsArray SqlStorage::BoundStatement::raw(jsg::Lock& js,
+    jsg::Optional<Statement::RawOptions> options) {
+  return statement->rawImpl(js, cloneBindings(), options.orDefault({}));
+}
+
+jsg::JsValue SqlStorage::BoundStatement::first(
+    jsg::Lock& js, jsg::Optional<jsg::JsString> column) {
+  return statement->firstImpl(js, cloneBindings(), column);
+}
+
+jsg::Arguments<SqlStorage::BindingValue> SqlStorage::BoundStatement::cloneBindings() {
+  // Annoyingly the bound statement could theoretically be executed multiple times and so we need
+  // to clone the bindings each time in order to pass an owned copy to the query while also keeping
+  // a copy for future calls. In theory with some refactoring we could avoid this but this is only
+  // a compatibility shim anyway so we're not optimizing for it.
+
+  return KJ_MAP(binding, bindings) -> BindingValue {
+    KJ_IF_SOME(b, binding) {
```

Comment:
Why not binding.map(...) here since that would return `kj::none` ? Just curious.